src/Database/LSMTree/Internal/Snapshot.hs

```diff
@@ -49,87 +58,49 @@ import           Database.LSMTree.Internal.UniqCounter (UniqCounter,
                      incrUniqCounter, uniqueToRunNumber)
 import           System.FS.API (HasFS)
 import           System.FS.BlockIO.API (HasBlockIO)
+import           Text.Printf
 
 {-------------------------------------------------------------------------------
   Versioning
 -------------------------------------------------------------------------------}
 
 -- | The version of a snapshot.
 --
--- When encoding snapshot metadata, 'currentSnapshotVersion' is included in the
--- file. When snapshot metadata is decoded, then the decoded version is checked
--- for compatibility against 'currentSnapshotVersion'. Decoding will fail if the
--- versions are incompatible.
---
--- A version @x.y@ has two components: a major version number @x@ and a minor
--- version number @y@. The major version number is used to signal breaking
--- changes. The minor version number is used to signal non-breaking changes that
--- are backwards compatible. To be precise, @x.y@ is guaranteed to be backwards
--- compatible with @x'.y'@ as long as @x == x'@ and @y' <= y@. If @x > x'@, then
--- backwards compatibility *may* be provided, but is not guaranteed. Forward
--- compatibilitty is never guaranteed.
---
--- If @x.y@ is our current version, and if it @x.y@ is backwards compatible with
--- @x'.y'@, then @x.y@ can decode snapshots that were encoded at version
--- @x'.y'@.
---
--- The version number determines the format of the snapshot metadata file, but
--- it also doubles as versioning information for the entire snapshot itself. For
--- example, if a breaking change is made to the merge scheduling algorithm that
--- would lead to errors when loading an older snapshot, then the major version
--- should be increased as well.
-data SnapshotVersion = V0_0
+-- A snapshot format version has one component: a version number @x@. A single
```

Comment:
A simpler way of saying this would be "a snapshot format version is a number", maybe also mentioning they're consecutive and increasing?